### PR TITLE
fix(aws-amplify-react): added prop onS3ImageSuccess to the S3Image component to have a clear separation of concerns

### PR DESCRIPTION
**Pull request by dudenamedjune**
fix(aws-amplify-react): added prop onS3ImageSuccess to the S3Image component to have a clear separation of concerns

_Issue #, if available:_
https://github.com/aws-amplify/amplify-js/issues/4847
_Description of changes:_

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by dudenamedjune** [comment]
Seems like there were some outdated snapshots in the auth component

File: packages/aws-amplify-react/src/Storage/S3Image.tsx [hunk 1]
```diff
@@ -36,6 +36,7 @@ export interface IS3ImageProps {
 	onClick?: any;
 	onError?: any;
 	onLoad?: any;
+	onS3ImageSuccess?: any;
```
Comment (jordanranz) [inline packages/aws-amplify-react/src/Storage/S3Image.tsx]: The `S3Image` component does both uploading and retrieval from S3. Can this variable be renamed to better reflect where the callback is executing. Something like `onUploadSuccess` or `onS3PutSuccess`. Also is there a better type we can apply to the function?

**Comment by jordanranz** [comment]
Thanks for contributing @dudenamedjune! Had a small comment above but otherwise looks good to me

**Comment by dudenamedjune** [comment]
@jordanranz Anytime! I changed the name, but as far as typing goes the other props in the interface have any for function only props. I'm not aware of a function type, but you can do something like this 

```js
export interface IS3ImageProps {
	onUploadSuccess?: () => {};
```

**Pull request merged**